common/djangoapps/student/models.py

```diff
@@ -332,6 +337,57 @@ def user_profile_pre_save_callback(sender, **kwargs):
         user_profile.profile_image_uploaded_at = None
 
 
+@receiver(post_save, sender=UserProfile)
+def user_profile_post_save_callback(sender, **kwargs):
+    """
+    Emit analytics events after saving the
+    """
+    user_profile = kwargs['instance']
+    # pylint: disable=protected-access
+    emit_field_changed_events(user_profile, user_profile.user, sender._meta.db_table)
+
+
+def emit_field_changed_events(instance, user, db_table):
+    """ For the given model instance, save the fields that have changed since the last save.
+
+    Requires that the Model uses 'model_utils.FieldTracker' and has assigned it to 'tracker'.
+
+    Args:
+        instance (Model instance): the model instance that is being saved
+        user (User): the user that this instance is associated with
+        db_table (str): the name of the table that we're modifying
+
+    Returns:
+        None
+    """
+    excluded_fields = ['meta', 'password']
+    changed_fields = instance.tracker.changed()
+    changed_settings = {}
+    for field in changed_fields:
+        if field not in excluded_fields:
+            field_dict = {
+                field: {
+                    'old_value': unicode(changed_fields[field]),
+                    'new_value': unicode(getattr(instance, field)),
+                }
+            }
+            changed_settings.update(field_dict)
+
+    # only emit events when something has changed
+    if len(changed_settings) > 0:
+        context = {'user_id': user.id}
```

Comment:
Must: Let the event tracking take care of the context for us. This should be the user who is actually making the change-- we don't want to force it to be the user whose model data is changing. Add a unit test for this case (one user changing profile info for another user, which can be done directly on the UserProfile object).

You can just do tracker.emit without the "with context" part.